implementation/src/main/java/io/smallrye/mutiny/helpers/queues/Queues.java

```diff
@@ -1,38 +1,53 @@
 package io.smallrye.mutiny.helpers.queues;
 
 import java.util.Queue;
-import java.util.concurrent.ArrayBlockingQueue;
 import java.util.function.Supplier;
 
-@SuppressWarnings({ "rawtypes", "unchecked" })
-public class Queues {
+import org.jctools.queues.atomic.MpscAtomicArrayQueue;
+import org.jctools.queues.atomic.MpscUnboundedAtomicArrayQueue;
+import org.jctools.queues.atomic.SpscAtomicArrayQueue;
+import org.jctools.queues.atomic.SpscChunkedAtomicArrayQueue;
+import org.jctools.queues.atomic.SpscUnboundedAtomicArrayQueue;
+import org.jctools.queues.unpadded.MpscUnboundedUnpaddedArrayQueue;
+import org.jctools.queues.unpadded.MpscUnpaddedArrayQueue;
+import org.jctools.queues.unpadded.SpscChunkedUnpaddedArrayQueue;
+import org.jctools.queues.unpadded.SpscUnboundedUnpaddedArrayQueue;
+import org.jctools.queues.unpadded.SpscUnpaddedArrayQueue;
 
-    /**
-     * Queues with a requested with a capacity greater than this value are unbounded.
-     */
-    public static final int TO_LARGE_TO_BE_BOUNDED = 10_000_000;
+import io.smallrye.mutiny.infrastructure.Infrastructure;
+
+public class Queues {
 
     private Queues() {
         // avoid direct instantiation
     }
 
-    public static final int BUFFER_XS = Math.max(8,
-            Integer.parseInt(System.getProperty("mutiny.buffer-size.xs", "32")));
-
-    public static final int BUFFER_S = Math.max(16,
-            Integer.parseInt(System.getProperty("mutiny.buffer-size.s", "256")));
-
-    static final Supplier EMPTY_QUEUE_SUPPLIER = EmptyQueue::new;
-    static final Supplier SINGLETON_QUEUE_SUPPLIER = SingletonQueue::new;
+    public static <T> Queue<T> createSpscArrayQueue(int size) {
+        if (Infrastructure.useUnsafeForQueues()) {
+            return new SpscUnpaddedArrayQueue<>(size);
+        } else {
+            return new SpscAtomicArrayQueue<>(size);
+        }
+    }
 
-    static final Supplier XS_QUEUE_SUPPLIER = () -> new SpscArrayQueue<>(BUFFER_XS);
-    static final Supplier S_QUEUE_SUPPLIER = () -> new SpscArrayQueue<>(BUFFER_S);
+    public static <T> Queue<T> createSpscUnboundedArrayQueue(int size) {
+        if (Infrastructure.useUnsafeForQueues()) {
+            return new SpscUnboundedUnpaddedArrayQueue<>(size);
+        } else {
+            return new SpscUnboundedAtomicArrayQueue<>(size);
+        }
+    }
 
-    static final Supplier UNBOUNDED_QUEUE_SUPPLIER = () -> new SpscLinkedArrayQueue<>(BUFFER_S);
-    static final Supplier XS_UNBOUNDED_QUEUE_SUPPLIER = () -> new SpscLinkedArrayQueue<>(BUFFER_XS);
+    public static <T> Queue<T> createSpscChunkedArrayQueue(int size) {
```

Comment:
This can (it's a choise) provide an alternative one with both allow to set:
- chunkSize
- capacity 

Taking in account that:
https://github.com/JCTools/JCTools/blob/463181b8dc147d045db62d2823c82c790abbca73/jctools-core/src/main/java/org/jctools/queues/SpscChunkedArrayQueue.java#L35C49-L35C57

the original heuristic can cause things to behave in a weird way if the user won't have any control on this